posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,322 @@
+
+== Test Transformation Strategy
+
+The stratgy used for Jakarta EE 9 test enablement in Open Liberty is to annotate test classes and methods to indicate which of these are to be run as Jakarta EE 9 tests.  The Open Liberty test framework transforms and re-runs the annotated tests as Jakarta EE 9 tests.
+
+Two types of updates are required when transforming test artifacts.  First, package references must be updated, changing old "javax" package references into new "jakarta" package references.  Package references can be in several locations, including test applications, test code, and server configurations.  Second, server configurations must be updated to replace old Jakarta EE 8 features with corresponding new Jakarta EE 9 features.
+
+Open Liberty updates package references using using the Eclipse Transformer.
+
+The Open Liberty test framework updates the feature names in the server configurations using existing function in the framework. Alternatively, the transformer could be used to update features in the server configuration if rules data were created with the substitution values needed for this update. Application and test developers will need to create code similar to the test framework code, or will need to update server configurations manually.
+
+== Invoking the Transformer
+
+The Eclipse Transformer can be used directly from the command line, as a build task, or from Java code.  This article describes using the transformer from the command line.
+
+(See the link:https://projects.eclipse.org/projects/technology.transformer[transformer project pages] for detailed usage guidelines.)
+
+=== Downloading the Transformer 
+
+Top level link:https://projects.eclipse.org/projects/technology.transformer/downloads[downloads links] are available within the project pages.
+
+Several current link:https://oss.sonatype.org/content/repositories/snapshots/org/eclipse/transformer/org.eclipse.transformer.cli/[snapshots] are usually available.
+
+=== Building the transformer
+
+Alternatively, the transformer can be built by cloning the link:https://github.com/eclipse/transformer[Eclipse Transformer repository] and invoking maven to build the transformer jar:
+
+[source]
+--
+    maven clean package
+--
+
+=== Preparing a work area
+
+Download or copy the transformer binaries to a local directory:
+
+[source]
+--
+    mkdir <work>
+    cd <work>
+    cp transformer/org.eclipse.transformer.cli/target/org.eclipse.transformer.cli-0.3.0-SNAPSHOT-distribution.jar .
+    unzip org.eclipse.transformer.cli-0.3.0-SNAPSHOT-distribution.jar
+--
+
+Unzipping the distribution obtains:
+
+[source]
+--
+    org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar
+    libs/org.eclipse.transformer-0.3.0-SNAPSHOT.jar
+    libs/biz.aQute.bnd.transform-5.2.0.jar
+    libs/slf4j-api-1.7.25.jar
+    libs/commons-cli-1.4.jar
+    libs/slf4j-simple-1.7.30.jar
+--
+
+=== Running the transformer
+
+The transformer can now be run using the default rules:
```

Comment:
The transformer can now be run using the default rules:
--->
To run the transformer by using the default rules, run the following command:

Reply:
Updated.  I don't like the sound of saying "run" twice, but I don't have an update which fixes this.